Adapters/Vungle/ISVungleAdapter/ISVungleInterstitialAdapterRouter.m

```diff
@@ -0,0 +1,116 @@
+//
+//  ISVungleInterstitialAdapterRouter.m
+//  ISVungleAdapter
+//
+//  Copyright © 2020 ironSource. All rights reserved.
+//
+
+#import "ISVungleInterstitialAdapterRouter.h"
+#import "ISVungleAdapter.h"
+
+@interface ISVungleInterstitialAdapterRouter()
+
+@property (nonatomic, strong, nullable) NSString *bidPayload;
+
+@end
+
+@implementation ISVungleInterstitialAdapterRouter
+
+- (instancetype)initWithPlacementID:(NSString *)placementID
+                      parentAdapter:(ISVungleAdapter *)parentAdapter
+                           delegate:(id<ISInterstitialAdapterDelegate>)delegate {
+    if (self = [super init]) {
+        _placementID = placementID;
+        _parentAdapter = parentAdapter;
+        _delegate = delegate;
+        _bidPayload = nil;
+    }
+
+    return self;
+}
+
+- (void)loadInterstitial {
+    self.interstitialAd = [[VungleInterstitial alloc] initWithPlacementId:self.placementID];
+    self.interstitialAd.delegate = self;
+    
+    if ([self.interstitialAd canPlayAd]) {
+        LogInternal_Internal(@"Interstitial ad: %@ is loaded", self.placementID);
+        [self.delegate adapterInterstitialDidLoad];
+        return;
+    }
```

Comment:
L36 - L40 is not needed here because there is no caching right now. So it will never be ready to play after making the new ad instance.

Reply:
+1

Reply:
@stanley-vungle @ManojBudumuru-vungle Thanks a lot for the guidance, updated and add some comments. :)